salt/orch/update.sls

```diff
@@ -1,3 +1,19 @@
+{# the number of etcd masters that should be in the cluster #}
+{%- set num_etcd_members = salt.caasp_etcd.get_cluster_size() %}
+{%- set additional_etcd_members = salt.caasp_etcd.get_additional_etcd_members() %}
+
+{% if additional_etcd_members|length > 0 %}
+# Mark some machines as new etcd members
+set-etcd-roles:
+  salt.function:
+    - tgt: {{ additional_etcd_members|join(',') }}
+    - tgt_type: list
+    - name: grains.append
+    - arg:
+      - roles
+      - etcd
+{% endif %}
+
 admin-apply-haproxy:
```

Comment:
Should we add a "require" to this state, so that if setting etcd roles fails, the upgrade doesn't proceed?

Reply:
Yes, will update.

Reply:
Done.